Mathlib/Data/Option/Basic.lean

```diff
@@ -426,4 +426,19 @@ theorem elim_comp₂ (h : α → β → γ) {f : γ → α} {x : α} {g : γ →
 theorem elim_apply {f : γ → α → β} {x : α → β} {i : Option γ} {y : α} :
     i.elim x f y = i.elim (x y) fun j => f j y := by rw [elim_comp fun f : α → β => f y]
 
+@[simp]
+theorem eq_none_or_exists_some_eq : ∀ x : Option α, x = none ∨ ∃ y, some y = x
```

Comment:
If I remember correctly, this lemma already exists for `∃ y, x = some y`, which is some small evidence in favor of #9968